cardano-api/src/Cardano/Api/Script.hs

```diff
@@ -749,6 +751,25 @@ toShelleyScript (ScriptInEra langInEra (SimpleScript _ script)) =
       SimpleScriptV2InAllegra -> toAllegraTimelock script
       SimpleScriptV2InMary    -> toAllegraTimelock script
 
+fromShelleyBasedScript  ::  forall era
+                        .   IsShelleyBasedEra era
+                        =>  Ledger.Script (ShelleyLedgerEra era)
+                        ->  ScriptInEra era
+fromShelleyBasedScript script =
+  case shelleyBasedEra @era of
```

Comment:
We've tended to conclude that for most functions like this things work out better in the end if we pass the era explicitly rather than implicitly via a type class argument.

i.e. like this (untested)
```suggestion
fromShelleyBasedScript  :: ShelleyBasedEra era
                        -> Ledger.Script (ShelleyLedgerEra era)
                        -> ScriptInEra era
fromShelleyBasedScript era script =
  case era of
```

Reply:
Yep the majority of the API we expose expects the era explicitly.

Reply:
done